include/zephyr/drivers/gpio.h

```diff
@@ -417,6 +417,240 @@ struct gpio_dt_spec {
 #define GPIO_DT_SPEC_INST_GET_OR(inst, prop, default_value) \
 	GPIO_DT_SPEC_INST_GET_BY_IDX_OR(inst, prop, 0, default_value)
 
+/*
+ * Auxiliary conditional macro that generates a bitmask for the range
+ * from @p "prop" array defined by the (off_idx, sz_idx) pair,
+ * or 0 if the range does not exist.
+ *
+ * @param node_id devicetree node identifier
+ * @param prop lowercase-and-underscores array property name
+ * @param off_idx logical index of bitmask offset value into "prop" array
+ * @param sz_idx logical index of bitmask size value into "prop" array
+ */
+#define GEN_BITMASK_COND(node_id, prop, off_idx, sz_idx)		     \
```

Comment:
Since this is a public header, please prefix the internal macros with `Z_NAMESPACE` (i.e. `Z_GPIO`) and exclude them from public API documentation using `@cond INTERNAL_HIDDEN`. This goes for all internal macros below. Only the user-facing macros should be included in the API documentation.

```suggestion
#define Z_GPIO_GEN_BITMASK_COND(node_id, prop, off_idx, sz_idx)		     \
```

Reply:
Done

Reply:
@henrikbrixandersen @cfriedt Sorry guys, but I'm waiting again for more then 2 week. @henrikbrixandersen Could you please finishing the review?

Reply:
Hi Andrey,

We are still waiting for you to implement the requested changes (use `Z_GPIO_` as prefix for internal macros, hide them from the API documentation using `@cond INTERNAL_HIDDEN`). Beware, we are currently in feature freeze for Zephyr v3.4.0, so people are focusing on fixing bugs prior to the release.

Reply:
Hi @henrikbrixandersen ,
Sorry, I've forget @cond  part of your suggestion. It has been done now, so we can unlock the merge process.